pyiron_base/jobs/job/runfunction.py

```diff
@@ -782,6 +782,86 @@ def multiprocess_wrapper(
         job_wrap.job.run_static()
 
 
+def generate_calculate_function(write_input_funct=None, collect_output_funct=None):
+    def calculate(input_dict, executable_dict, output_dict={}):
+        working_directory = executable_dict["working_directory"]
+        accept_crash = executable_dict.pop("accept_crash", False)
+        accepted_return_codes = executable_dict.pop("accepted_return_codes", [])
+        os.makedirs(working_directory, exist_ok=True)
+        if write_input_funct is not None:
+            write_input_funct(
+                input_dict=input_dict,
+                working_directory=working_directory,
+            )
+        try:
+            shell_output = execute_subprocess(**executable_dict)
+            error = None
+            job_crashed = False
+        except (subprocess.CalledProcessError, FileNotFoundError) as error:
+            if hasattr(error, "output"):
+                shell_output = error.output
+                if error.returncode in accepted_return_codes:
+                    job_crashed = False
+                elif not accept_crash:
+                    error_file = posixpath.join(working_directory, "error.msg")
+                    with open(error_file, "w") as f:
+                        f.write(error.output)
+                    raise RuntimeError("External executable failed")
+                else:
+                    job_crashed = True
+            else:
+                raise RuntimeError("Job aborted")
+        with open(posixpath.join(working_directory, "error.out"), mode="w") as f_err:
+            f_err.write(shell_output)
+        if (
+            not job_crashed
+            and collect_output_funct is not None
+            and len(output_dict) > 0
+        ):
+            parsed_output = collect_output_funct(
+                working_directory=working_directory,
+                **output_dict,
+            )
+        elif not job_crashed and collect_output_funct is not None:
+            parsed_output = collect_output_funct(
+                working_directory=working_directory,
+            )
+        else:
+            parsed_output = None
+        return shell_output, parsed_output, job_crashed
+
+    return calculate
+
+
+def execute_calculate_function(job):
```

Comment:
I like the direction of this, but as long as it's not general I'd like a warning or a comment, when `job` is not `ExecutableContainerJob`, because it's not obvious from the function itself.  Comment is probably fine, because no user will call this accidentally, but developers other than you will have a hard time without.

Reply:
The same function is used in the `ScriptJob` in https://github.com/pyiron/pyiron_base/pull/1480 and I plan to use it in LAMMPS and VASP once these changes are merged and a new version of `pyiron_base` is released.